Further adoption of the new CoreIPC serialization mechanism for RemoteMediaPlayerState.

sheeparegreat opened the pull request:
#### 31d68e08a6df7f911ec6647984d88885a264a291
<pre>
Further adoption of the new CoreIPC serialization mechanism for RemoteMediaPlayerState.
<a href="https://bugs.webkit.org/show_bug.cgi?id=247825.">https://bugs.webkit.org/show_bug.cgi?id=247825.</a>
rdar://102255168.

Reviewed by Alex Christensen.

* Source/WebCore/platform/graphics/PlatformTimeRanges.cpp:
(WebCore::PlatformTimeRanges::invert):
(WebCore::PlatformTimeRanges::unionWith):
(WebCore::PlatformTimeRanges::start const):
(WebCore::PlatformTimeRanges::end const):
(WebCore::PlatformTimeRanges::duration const):
(WebCore::PlatformTimeRanges::maximumBufferedTime const):
(WebCore::PlatformTimeRanges::add):
(WebCore::PlatformTimeRanges::copyWithEpsilon const):
* Source/WebCore/platform/graphics/PlatformTimeRanges.h:
(WebCore::PlatformTimeRanges::encode const): Deleted.
(WebCore::PlatformTimeRanges::decode): Deleted.
* Source/WebCore/platform/graphics/VideoPlaybackQualityMetrics.h:
(WebCore::VideoPlaybackQualityMetrics::encode const): Deleted.
(WebCore::VideoPlaybackQualityMetrics::decode): Deleted.
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.h:
(WebKit::RemoteMediaPlayerState::encode const): Deleted.
(WebKit::RemoteMediaPlayerState::decode): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.serialization.in: Added.

Canonical link: <a href="https://commits.webkit.org/256733@main">https://commits.webkit.org/256733@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c144fb34fe2ad66e33e1b6d921f46d33ce5ff701

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96598 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5854 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29675 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106124 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/166459 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100582 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6050 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34592 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88972 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/102835 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102273 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/4513 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83201 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/31481 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/86356 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/88235 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/74391 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/40323 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19726 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37986 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21129 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/4255 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/43670 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2236 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1068 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/40405 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

philn commented on Source/WebKit/CMakeLists.txt:
```diff
@@ -611,6 +611,8 @@ set(WebKit_SERIALIZATION_IN_FILES
 
     Shared/WebsiteData/WebsiteDataFetchOption.serialization.in
+
+    WebProcess/GPU/media/RemoteMediaPlayerState.serialization.in
```
So this now implies GPU_PROCESS has to be enabled in Linux ports, which is not the case, which is the cause of the build failures.

RemotePlayerMediaState.h has no GPU_PROCESS specific code, so I think we could remove the ifdef guards in that file and that would make EWS happy.

zdobersek replied:
Or add the `ENABLE(GPU_PROCESS)` guards to the serialization file.

sheeparegreat committed:
Further adoption of the new CoreIPC serialization mechanism for RemoteMediaPlayerState.
https://bugs.webkit.org/show_bug.cgi?id=247825.
rdar://102255168.

Reviewed by Alex Christensen.

* Source/WebCore/platform/graphics/PlatformTimeRanges.cpp:
(WebCore::PlatformTimeRanges::invert):
(WebCore::PlatformTimeRanges::unionWith):
(WebCore::PlatformTimeRanges::start const):
(WebCore::PlatformTimeRanges::end const):
(WebCore::PlatformTimeRanges::duration const):
(WebCore::PlatformTimeRanges::maximumBufferedTime const):
(WebCore::PlatformTimeRanges::add):
(WebCore::PlatformTimeRanges::copyWithEpsilon const):
* Source/WebCore/platform/graphics/PlatformTimeRanges.h:
(WebCore::PlatformTimeRanges::encode const): Deleted.
(WebCore::PlatformTimeRanges::decode): Deleted.
* Source/WebCore/platform/graphics/VideoPlaybackQualityMetrics.h:
(WebCore::VideoPlaybackQualityMetrics::encode const): Deleted.
(WebCore::VideoPlaybackQualityMetrics::decode): Deleted.
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.h:
(WebKit::RemoteMediaPlayerState::encode const): Deleted.
(WebKit::RemoteMediaPlayerState::decode): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.serialization.in: Added.

Canonical link: https://commits.webkit.org/256733@main
